sdk/src/main/scala/com/horizen/api/http/client/SecureEnclaveApiClient.scala

```diff
@@ -0,0 +1,88 @@
+package com.horizen.api.http.client
+
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import akka.http.scaladsl.client.RequestBuilding.Post
+import akka.http.scaladsl.model.{HttpEntity, HttpRequest, HttpResponse, StatusCodes}
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.http.scaladsl.{Http, HttpExt}
+import akka.pattern.pipe
+import com.horizen.RemoteKeysManagerSettings
+import com.horizen.api.http.JacksonSupport._
+import com.horizen.api.http.client.SecureEnclaveApiClient.{CreateSignatureRequest, CreateSignatureResponse, SignWithEnclave}
+import com.horizen.certificatesubmitter.CertificateSubmitter.CertificateSignatureInfo
+import com.horizen.proof.SchnorrSignatureSerializer
+import com.horizen.proposition.{SchnorrProposition, SchnorrPropositionSerializer}
+import com.horizen.utils.BytesUtils
+import io.circe.generic.auto._
+import io.circe.syntax._
+import scorex.util.ScorexLogging
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SecureEnclaveApiClient(settings: RemoteKeysManagerSettings)(implicit ec: ExecutionContext) extends Actor with ScorexLogging {
```

Comment:
Is there any benefit of having SecureEnclaveApiClient as an Actor.
You may look at `mainchainChannel: MainchainNodeChannel` and it's usage/implementations.

I think that it is more than enough to define `trait SecureEnclaveApi`
with a couple of methods to know if it is enabled and to generate signatures.

It will reduce code complexity and will make it more stable.
Let's discuss it.